mesh/test/immersed_boundary/TestImmersedBoundaryFftInterface.hpp

```diff
@@ -0,0 +1,154 @@
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice,
+   this list of conditions and the following disclaimer.
+ * Redistributions in binary form must reproduce the above copyright notice,
+   this list of conditions and the following disclaimer in the documentation
+   and/or other materials provided with the distribution.
+ * Neither the name of the University of Oxford nor the names of its
+   contributors may be used to endorse or promote products derived from this
+   software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT
+OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+*/
+
+#ifndef TESTFLUIDSOURCE_HPP_
+#define TESTFLUIDSOURCE_HPP_
+
+// Needed for test framework
+#include <cxxtest/TestSuite.h>
+
+// Includes from projects/ImmersedBoundary
+#include "ImmersedBoundaryFftInterface.hpp"
+
+#include <complex>
+
+// This test is never run in parallel
+#include "FakePetscSetup.hpp"
+
+class TestImmersedBoundaryFftInterface : public CxxTest::TestSuite
+{
+public:
+
+    void TestConstructor()
+    {
+        // Create a square test mesh
+        std::vector<Node<2>*> nodes;
+        nodes.push_back(new Node<2>(0, true, 0.0, 0.0));
+        nodes.push_back(new Node<2>(1, true, 0.1, 0.0));
+        nodes.push_back(new Node<2>(2, true, 0.1, 0.1));
+        nodes.push_back(new Node<2>(3, true, 0.0, 0.1));
+
+        std::vector<ImmersedBoundaryElement<2, 2>*> elems;
+        elems.push_back(new ImmersedBoundaryElement<2, 2>(0, nodes));
+        elems.push_back(new ImmersedBoundaryElement<2, 2>(1, nodes));
+        elems.push_back(new ImmersedBoundaryElement<2, 2>(2, nodes));
+
+        ImmersedBoundaryMesh<2,2>* p_mesh = new ImmersedBoundaryMesh<2, 2>(nodes, elems);
+        
+        std::vector<double> in = {1.0, 0.5, 0.0, 0.5};
+        std::vector<std::complex<double>> comp(in.size());
+        std::vector<double> out(in.size());
+
+        ImmersedBoundaryFftInterface<2> fft(p_mesh, in.data(), comp.data(), out.data(), true);
+        TS_ASSERT_EQUALS(fft.mRealDims[0], p_mesh->GetNumGridPtsX());
+
+        delete p_mesh;
+    }
+
+    void TestFftExecuteForward()
```

Comment:
- Generate some test data using python/matlab, i.e. alternative implementation. 
- Add assertions to check that parameters such as strides are correct - ensure the interfacing with the library is correct.